cmake/macos_build_environment

```diff
@@ -0,0 +1,2 @@
+2.3-55d9a17-sdk10.15-macosminimum10.12-x86_64
+86c56d3aba24ec23923ad906f5a13084b0f784c955812448ecb43821fd907d50
```

Comment:
Couldn't we just run sha256sum when building the environment and deploy a second file `<buildenvname>.tar.gz.sha256sum` to the server? That's the standard way to do this IMHO. 

Then we can run `sha256sum -c` locally which would simplify the code and we won't have to maintain the checksum in the Mixxx repo.

Reply:
I think it's better to keep this in the code repository. That way we can know what build environment was used for any given build.

Reply:
This also allows us to revert to an older build environment if something goes wrong with a new one.

Reply:
This is just about the sha256 sum. I agree we should keep the buildenv name in the repository.

Reply:
If you want to implement that, go ahead. This works fine though.